deploy/cr/cr_cloud_auth.yaml

```diff
@@ -0,0 +1,19 @@
+# This define an Infinispan cluster that wants app and mgmt user/pass
+# specified in a secret. If the secret's name is specified the secret
+# must contains all the authentication keys:
+#       APP_USER
+#       APP_PASS
+#       MGMT_USER
+#       MGMT_PASS
+# the cr_cloudsecret.yaml file is an example and can be used like this:
+# oc apply -f cloud_secret.yaml
+#
+
+apiVersion: infinispan.org/v1
+kind: Infinispan
+metadata:
+  name: example-infinispan
+cloud:
```

Comment:
Why is `cloud` outside of `spec`? When does it make sense to do that? Couldn't easily find info on this.

Reply:
`cloud` should be in `spec`, I've tracked that task in a separate issue.  See discussion above with Crumby.

Actually the _status_ of an Infinispan object is something undefined to me, but I don't it's a problem atm.

The only design choice here is to put under `cloud` all the setup data that works with the cloud.xml configuration.

Reply:
> The only design choice here is to put under cloud all the setup data that works with the cloud.xml configuration.

Not sure I agree with that. Although we allow cloud XML file to be provided by the user, we want to move towards a situation where we don't expose that at all, hence I don't even think calling it `cloud` makes sense. 

I'm happy to leave it as `cloud`, but in the future it might not indicative. Something like `server` or `config` might be more suitable.